x/auth/tx/config.go

```diff
@@ -34,6 +34,26 @@ func NewTxConfig(protoCodec codec.ProtoCodecMarshaler, enabledSignModes []signin
 	}
 }
 
+// CustomTxConfig returns a new protobuf TxConfig using the provided ProtoCodec, sign modes, and (de/en)coders. The
+// first enabled sign mode will become the default sign mode.
+func CustomTxConfig(
+	handler signing.SignModeHandler,
+	decoder sdk.TxDecoder,
+	encoder sdk.TxEncoder,
+	jsonDecoder sdk.TxDecoder,
+	jsonEncoder sdk.TxEncoder,
+	protoCodec codec.ProtoCodecMarshaler,
```

Comment:
I can see wanting to add additional sign modes, but overloading all the decoders and encoders seems like a smell here.